### PR TITLE
Refactory/repository dependency injection

### PR DESCRIPTION
**Pull request by BrainMaestro**
Don't be scared by the big diff. We need to move quickly with this one.

### CODE REVIEW

**Comment by tenorz007** [comment]
You should know, this is a lot..... you would have at least separated them by commits, it would be a lot easier to follow.

**Comment by BrainMaestro** [comment]
Well I did or tried to. It's mostly the same change repeated across the different repositories and their tests. That's why it looks like a lot. It didn't seem right to split the same change across different commits. 

The basic idea was to stop hardcoding the `Model` and `Transformer`. They are being injected instead.

**Comment by BrainMaestro** [comment]
Use split view. Makes it easier to see the actual difference.

Approved!

File: app/Repositories/Database/SQLKeyRepository.php [hunk 1]
```diff
@@ -108,15 +103,13 @@ public function setUsed($id)
...
      * @param $sectionId
-     * @return bool
+     * @return boolean
      */
     public function deleteKeys($sectionId)
     {
-        $this->model->where('section_id', $sectionId)->delete();
-
-        return true;
+        return $this->model->where('section_id', $sectionId)->delete() > 0;
```
Comment (tenorz007) [inline app/Repositories/Database/SQLKeyRepository.php]: What's this "> 0"?

Comment (BrainMaestro) [inline app/Repositories/Database/SQLKeyRepository.php]: Delete returns an `int` for the number of deleted columns. If it's 0, that means the delete failed.

**Comment by BrainMaestro** [comment]
This is waiting for your approval

**Comment by tenorz007** [comment]
Approved!

**Pull request merged**